site/app/entities/poll/Poll.php

```diff
@@ -91,39 +98,74 @@ public function getQuestion(): string {
     public function setQuestion(string $question): void {
         $this->question = $question;
     }
-
-    public function getStatus(): string {
-        return $this->status;
+    public function setClosed(): void {
+        $this->end_date = new \DateTime(DateUtils::BEGINING_OF_TIME);
     }
-
     public function setOpen(): void {
-        $this->status = "open";
+        $this->end_date = new \DateTime(DateUtils::MAX_TIME);
+    }
+    public function setEnded(): void {
+        $temp = DateUtils::getDateTimeNow();
+        $tempString = $temp->format('Y-m-d');
+        $this->end_date = new DateTime($tempString);
     }
-
     public function isOpen(): bool {
-        return $this->status === "open";
+        $now = DateUtils::getDateTimeNow();
+        return $now < $this->end_date;
     }
 
-    public function setClosed(): void {
-        $this->status = "closed";
+    public function isEnded(): bool {
+        $now = DateUtils::getDateTimeNow();
+        $closeDate = DateUtils::BEGINING_OF_TIME;
+        return $now > $this->end_date && $this->end_date->format('Y-m-d\TH:i:s') !== $closeDate;
     }
 
     public function isClosed(): bool {
-        return $this->status === "closed";
+        $now = DateUtils::getDateTimeNow();
+        return $now > $this->end_date && $this->end_date->format('Y-m-d\TH:i:s') === DateUtils::BEGINING_OF_TIME;
     }
 
-    public function setEnded(): void {
-        $this->status = "ended";
+    public function getDuration(): \DateInterval {
+        $seconds = $this->duration;
+        $years = floor($seconds / (365 * 24 * 60 * 60));
+        $seconds -= $years * (365 * 24 * 60 * 60);
+
+        $months = floor($seconds / (30 * 24 * 60 * 60));
+        $seconds -= $months * (30 * 24 * 60 * 60);
+
+        $days = floor($seconds / (24 * 60 * 60));
+        $seconds -= $days * (24 * 60 * 60);
+
+        $hours = floor($seconds / (60 * 60));
+        $seconds -= $hours * (60 * 60);
+
+        $minutes = floor($seconds / 60);
+        $seconds -= $minutes * 60;
+        return new \DateInterval("P{$years}Y{$months}M{$days}DT{$hours}H{$minutes}M{$seconds}S");
     }
 
-    public function isEnded(): bool {
-        return $this->status === "ended";
+    public function getEndDate(): \DateTime {
+        return $this->end_date;
     }
 
     public function getReleaseDate(): \DateTime {
         return $this->release_date;
     }
 
+    public function setDuration(\DateInterval $duration): void {
+        $totalSeconds = $duration->s;
+        $totalSeconds += $duration->i * 60;
+        $totalSeconds += $duration->h * 3600;
+        $totalSeconds += $duration->d * 86400;
+        $totalSeconds += $duration->m * 2592000;
+        $totalSeconds += $duration->y * 31536000;
+        $this->duration = $totalSeconds;
+    }
+
+    public function setEndDate(\DateTime $end_date): void {
+        $this->end_date = $end_date;
+    }
+
```

Comment:
Again, we should also simplify the setting of a poll's duration.

Reply:
The main reason for making it a DateInterval is the ease of access where you can add a DateInterval directly to a DateTime Object.